src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RandomNumberGenerator.cs

```diff
@@ -96,12 +97,107 @@ public static void Fill(Span<byte> data)
             RandomNumberGeneratorImplementation.FillSpan(data);
         }
 
+        public static int GetInt32(int fromInclusive, int toExclusive)
+        {
+            if (fromInclusive >= toExclusive) throw new ArgumentException(SR.Argument_InvalidRandomRange);
+
+            // The total possible range is [0, 4,294,967,295).
+            long adjustedRange = (long)toExclusive - fromInclusive;
+
+            // If there is only one possible choice, nothing random will actually happen, so return
+            // the only possibility.
+            if (adjustedRange == 1)
+            {
+                return fromInclusive;
+            }
+
+            // Subtract one to account for zero being an actual possibility. We know for certain that
+            // the range fits in a UInt32 without overflowing.
+            uint maskedRange = (uint)adjustedRange - 1;
+
+            int leadingZeroCount = LeadingZeroCount(maskedRange);
+
+            // Create a mask for the bits that we care about for the range. The other bits will be
+            // masked away.
+            uint mask;
+
+            // If the LZCNT is zero create a full mask to avoid overflowing when shifting.
+            if (leadingZeroCount == 0)
+            {
+                mask = unchecked((uint)-1);
+            }
+            else
+            {
+                mask = (1U << 32 - leadingZeroCount) - 1;
+            }
+
+            // We only want to generate as many bytes as required to satisfy the mask to not apply
+            // undue pressure to the underlying random number generator.
+            int bytesRequired = (32 + 7 - leadingZeroCount) / 8;
+
+            Span<uint> valueResult = stackalloc uint[1];
+            ref uint result = ref valueResult[0];
+            Span<byte> valueResultBytes = MemoryMarshal.AsBytes(valueResult);
+            if (BitConverter.IsLittleEndian)
```

Comment:
My personal style is to have a blank line before a control flow statement because I find it helps legibility. I'm just offering it out there, our style guide isn't prescriptive on this one way or another.